### PR TITLE
Add /quit command

### PR DESCRIPTION
**Pull request by trevarj**
I often see users in the IRC channel wondering how to quit. They usually figure it out, but I think `/quit` is universal among tui IRC clients

### CODE REVIEW

**Comment by osa1** [comment]
Thanks @trevarj .

I think `/quit` shouldn't ask for confirmation. The reason why `C-c` asks for confirmation is because `C-c` is easy to press accidentally, and I think most users would have the muscle memory for it as it's used in shells. If you typed `/quit` you've already made some effort typing all the characters so it's probably not accidental and you really want to quit.

I didn't check other clients. Do they ask for confirmation when you type `/quit`?

**Comment by trevarj** [comment]
@osa1 Good point. I think it should quit automatically too. Not sure about the other clients

**Comment by trevarj** [comment]
Updated to quit immediately.

File: crates/libtiny_tui/src/tui.rs [hunk 1]
```diff
@@ -264,9 +273,10 @@ impl TUI {
                     );
                 }
                 // false to fall through to print help for cmd.rs commands
```
Comment (osa1) [inline crates/libtiny_tui/src/tui.rs:275]: Would be good to update this comment as we no longer return `false`.

File: crates/libtiny_tui/src/tui.rs [hunk 2]
```diff
@@ -90,6 +93,12 @@ pub struct TUI {
     config_path: Option<PathBuf>,
 }
 
+pub(crate) enum CmdResult {
+    Ok,
+    Skip,
+    Quit,
+}
```
Comment (osa1) [inline crates/libtiny_tui/src/tui.rs:100]: Would be good to document these variants. As far as I understand:

- `Ok`: command is handled by the TUI, don't propagate it to the caller.
- `Skip`: Command is not handled by the TUI, propagate it to the caller.
- `Quit`: Quit requested

If you document it this way then it becomes clear that "skip" is a bit confusing. It feels like the receiver (whoever got the `Skip` value) should skip, but actually it means the sender (whoever created the `Skip` value) is skipped so the receiver should handle it, right? Perhaps it should be called `Ignored` or `Skipped`.

**Review by osa1 — APPROVED** [review]
Thanks @trevarj. LGTM. Added some inline comments.

File: crates/libtiny_tui/src/tui.rs (hunk 2 above)
Comment (trevarj) [reply crates/libtiny_tui/src/tui.rs:103]: Changed to `Continue`

File: crates/libtiny_tui/src/lib.rs [hunk 3]
```diff
@@ -178,9 +178,17 @@ async fn input_handler<S>(
...
                             // Handle TUI commands, send others to downstream
                             let cmd: String = (&msg[1..]).iter().collect();
-                            let handled = tui.borrow_mut().try_handle_cmd(&cmd, &from);
-                            if !handled {
-                                snd_ev.try_send(Event::Cmd { cmd, source: from }).unwrap();
+                            let result = tui.borrow_mut().try_handle_cmd(&cmd, &from);
+                            match result {
+                                CmdResult::Ok => {}
+                                CmdResult::Continue => {
+                                    snd_ev.try_send(Event::Cmd { cmd, source: from }).unwrap()
```
Comment (trevarj) [inline crates/libtiny_tui/src/lib.rs:185]: When we get `Continue`, pass the command down to the client (cmd.rs) to handle it as well.

**Pull request merged**